src/Data/List/Relation/Binary/Permutation/Propositional.agda

```diff
@@ -89,12 +130,12 @@ module PermutationReasoning where
   -- Skip reasoning on the first element
   step-prep : ∀ x xs {ys zs : List A} → (x ∷ ys) IsRelatedTo zs →
               xs ↭ ys → (x ∷ xs) IsRelatedTo zs
-  step-prep x xs rel xs↭ys = relTo (trans (prep x xs↭ys) (begin rel))
+  step-prep x xs rel xs↭ys = ↭-go (↭-prep x xs↭ys) rel
```

Comment:
Again: legacy code.
All I have done here is encapsulate the permutation constructors and that of the underlying representing type `IsRelatedTo`, so that the appeal to `Reasoning.Syntax` is better encapsulated. 
NB. *no* connection here between `Homogeneous.steps` and the `step-*` combinators defined: the latter refers to steps of `Reasoning.Syntax`!

Reply:
Ok, happy to not devolve this into mission creep.